tripy/docs/pre0_user_guides/02-compiler.md

```diff
@@ -0,0 +1,169 @@
+# Using the Compiler
+
+```{contents} Table of Contents
+:depth: 3
+```
+
+## Walk through model compilation and deployment
+
+Let's walk through a simple example of a [GEGLU](https://arxiv.org/abs/2002.05202v1) module defined below:
+
+```py
+# doc: no-eval
+class GEGLU(tp.Module):
+    def __init__(self, dim_in, dim_out):
+        self.proj = tp.Linear(dim_in, dim_out * 2)
+        self.dim_out = dim_out
+    
+    def __call__(self, x):
+        proj = self.proj(x)
+        x, gate = tp.split(proj, 2, proj.rank - 1)
+        return x * tp.gelu(gate)
+```
+
+To naively run `GEGLU` in eager mode:
+
+```py
+# doc: no-eval
+layer = GEGLU(2, 8)
+inp = tp.ones((1, 2))
+out = layer(inp)
+print(out)
+```
+
+Now, let's try to optimize this model for inference using Tripy's Compiler.
+
+First, let's initialize the compiler with `layer` which lets the `Compiler` know the properties of the function we are compiling (e.g. function signature).
```

Comment:
Could we write it as either just:
```
compiler
```
Or:
```
{class}`Compiler`
```

The latter will link to the API docs